projects/buttons_and_adc/src/main.c

```diff
@@ -0,0 +1,61 @@
+// Created by Sudhish M on 2020-10-04.
+// buttons_and_adc
+
+#include "adc.h"
+#include "gpio.h"
+#include "gpio_it.h"
+#include "interrupt.h"
+#include "log.h"
+
+static GpioAddress adc_pin_address{ .port = GPIO_PORT_A, .pin = 6 };
+static GpioAddress button_address{ .port = GPIO_PORT_B, .pin = 2 };
+
+static GpioSettings button_settings = {
+  .direction = GPIO_DIR_IN,
+  .resistor = PIO_RES_PULLDOWN,
+  // Reason for using pulldown: when no signal is received
+  // from the button (unpressed), the value sent to the pin is 0
```

Comment:
You have a typo here: `PIO_RES_PULLDOWN` will need to be `GPIO_RES_PULLDOWN` to compile. Also generally the button will pull the signal down automatically when it's unpressed, but good thought! (And of course, remember the `s_` on `button_settings`.)